### PR TITLE
Define missing crypto.constants defined on Node

### PR DESCRIPTION
**Pull request by saklani**
### What does this PR do?

Some crypto constants are missing that are defined by Node in MacOS, Linux, Windows.
#8831
- [x] Code changes

### How did you verify your code works?
I am on an older version of MacOS (Darwin 21.6.0 x86_64 i386) so I can't build the latest version of bun, but I built an older version `v1.0.30`.

### Simple Replication across Platforms
Seem to be consistent across platform I tried so should be safe define.
Darwin 21.6.0 x86_64 i386 Node v21.6.0
Microsoft Windows NT 10.0.22631.0 x64 Node v20.11.0
Linux 6.1.77-99.164.amzn2023.x86_64 x86_64 x86_64 Node v18.18.2

#### Script
```js
import crypto from "crypto";

console.log(crypto.constants.RSA_X931_PADDING);
console.log(crypto.constants.RSA_PSS_SALTLEN_DIGEST);
console.log(crypto.constants.RSA_PSS_SALTLEN_MAX_SIGN);
console.log(crypto.constants.RSA_PSS_SALTLEN_AUTO);
```
Node Output
```sh
5
-1
-2
-2
```

Bun Output (v1.0.30) built on Darwin 21.6.0 x86_64 i386
```sh
5
-1
-2
-2
```

### Tests
Waiting for maintainer comment on if this requires a test.

### CODE REVIEW

File: src/bun.js/bindings/ProcessBindingConstants.cpp [hunk 1]
```diff
@@ -905,20 +905,20 @@ static JSValue processBindingConstantsGetCrypto(VM& vm, JSObject* bindingObject)
 #ifdef RSA_PKCS1_OAEP_PADDING
     object->putDirect(vm, PropertyName(Identifier::fromString(vm, "RSA_PKCS1_OAEP_PADDING"_s)), jsNumber(RSA_PKCS1_OAEP_PADDING));
 #endif
-#ifdef RSA_X931_PADDING
-    object->putDirect(vm, PropertyName(Identifier::fromString(vm, "RSA_X931_PADDING"_s)), jsNumber(RSA_X931_PADDING));
+#ifndef RSA_X931_PADDING
+    object->putDirect(vm, PropertyName(Identifier::fromString(vm, "RSA_X931_PADDING"_s)), jsNumber(5));
 #endif
```
Comment (jdalton) [inline src/bun.js/bindings/ProcessBindingConstants.cpp:912]: Thank you @saklani! We still want to use the values if they exist so we shouldn't remove the `#ifdef` but we should add an `#else` and put the default value.

**Review by paperclover — CHANGES_REQUESTED** [review]
see john-david's comment, but i think a better approach to this would be to find where these constants are defined. it is possible this is fixed by adding one missing `#include` at the top of this file.

File: src/bun.js/bindings/ProcessBindingConstants.cpp (hunk 1 above)
Comment (saklani) [reply src/bun.js/bindings/ProcessBindingConstants.cpp:912]: Done!

**Comment by saklani** [comment]
> see john-david's comment, but i think a better approach to this would be to find where these constants are defined. it is possible this is fixed by adding one missing `#include` at the top of this file.

I was hoping for the same. It would be a cleaner approach. It's not defined unfortunately.
Node's dependency of OpenSSL seems to define it. 

https://github.com/nodejs/node/blob/f4a0a3b04b92596fd2cf40e1d9c5a003e81c77b8/deps/openssl/openssl/include/openssl/rsa.h#L189

And from what I understand bun uses boringSSL instead it seems to not be defined there.

**Review by paperclover — APPROVED** [review]

**Review by jdalton — APPROVED** [review]

**Pull request merged**